stdlib/public/Concurrency/CooperativeGlobalExecutor.inc

```diff
@@ -136,14 +133,40 @@ static void swift_task_enqueueGlobalWithDelayImpl(JobDelay delay,
   *position = newJob;
 }
 
+/// Insert a job into the cooperative global queue with a delay.
+SWIFT_CC(swift)
+static void swift_task_enqueueGlobalWithDelayImpl(JobDelay delay,
+                                                  Job *newJob) {
+  assert(newJob && "no job provided");
+
+  auto deadline = std::chrono::steady_clock::now()
+                + std::chrono::duration_cast<JobDeadline::duration>(
+                    std::chrono::nanoseconds(delay));
+  JobDeadlineStorage<>::set(newJob, deadline);
+
+  insertDelayedJob(newJob, deadline);
+}
+
 SWIFT_CC(swift)
 static void swift_task_enqueueGlobalWithDeadlineImpl(long long sec,
                                                      long long nsec,
                                                      long long tsec,
                                                      long long tnsec,
-                                                     int clock, Job *job) {
-  assert(job && "no job provided");
-  // TODO: implementation
+                                                     int clock, Job *newJob) {
+  assert(newJob && "no job provided");
+
+  long long nowSec;
+  long long nowNsec;
+  swift_get_time(&nowSec, &nowNsec, (swift_clock_id)clock);
+
+  uint64_t delta = (sec - nowSec) * NSEC_PER_SEC + nsec - nowNsec;
+
+  auto deadline = std::chrono::steady_clock::now()
```

Comment:
So I guess this implementation just ignores the clock parameter then, since this does not have a suspending style clock (std::chrono only has continuous)

I guess that makes sense w/ a single threaded runtime.